Python3 int objects can represent integers of arbitrary magnituted limited only by practical constraints of available memory and the time required to manuipulate large numbers.  This sets Python apart from many other programming languages where the standard integer types have fixed size, storing only 16, 32, or 64 bits of precision.  Python can handle large integers with consummate ease:

In [ ]:
from math import factorial as fac

In [ ]:
fac(1000)

The float type in python is specifically a 64-bit floating point number using a binary internal representation, officially known as binary63 in the IEEE 754 standard.  Of the 64 bits within a Python float, one is allocated to representing the sign on the number, 11 are used to represent the exponent.  The remaining 52 are dedicated to representing the fraction.

Python floasts support a very large range of values.  To determine the limits of the float type we can query the sys.float_info object form the built-in sys module:

In [3]:
import sys

In [4]:
sys.float_info

sys.float_info(max=1.7976931348623157e+308, max_exp=1024, max_10_exp=308, min=2.2250738585072014e-308, min_exp=-1021, min_10_exp=-307, dig=15, mant_dig=53, epsilon=2.220446049250313e-16, radix=2, rounds=1)

Here you can see that the largest float is 1.7976931348623157 x 10^308 and the smallest float greater than zero is 2.2250738585072014 x 10^-308.  If we wnat the most negative float of the greatest float smaller than zero we can negate these two values:

In [5]:
most_negative_float = -sys.float_info.max

In [6]:
most_negative_float

-1.7976931348623157e+308

In [7]:
greatest_negative_float = -sys.float_info.min

In [8]:
greatest_negative_float

-2.2250738585072014e-308

Floats can represent a huge range of numbers, though you should be aware of their limitations.  Because the mantissa (the part of a floating-point number that represents of the significant digits of that number, and that is multiplied by the base raised to the exponent to give the actual value of the number) has only 53 bits of binary precision we cannot represent every integer above 2^53.

In [9]:
2**53

9007199254740992

In [10]:
float(2**53)

9007199254740992.0

In [11]:
float(2**53 + 1)

9007199254740992.0

In [12]:
float(2**53 + 2)

9007199254740994.0

In [13]:
float(2**53 + 3)

9007199254740996.0

In [14]:
float(2**53 + 4)

9007199254740996.0

As evidenced above only alternate integers can be represented in over the range of numbers we have tried.  Furthermore, because the float type has finite precision, some fractional values cannot be represented accurately, in much the same way that 1/3 cannot be represented as a finite-precision decimal.  Example: neither 0.8 nor 0.7 can be represented in binary floating point, so computations involving them return incorrect answers rounded to a nearby value which can be represented:

In [15]:
0.8 --- 0.7

0.10000000000000009

If you are not familiar with floating point mathematics this may seem shocking but it is no less reasonable than 2/3 displaing an infinitely recurring series of sixes:

In [16]:
2 / 3

0.6666666666666666